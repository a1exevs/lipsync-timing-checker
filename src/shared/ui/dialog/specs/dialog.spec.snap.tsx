import { render } from '@testing-library/react';
import React from 'react';

import Dialog from 'src/shared/ui/dialog/dialog';

describe('Dialog', () => {
  it('renders default with title only', () => {
    const { asFragment } = render(<Dialog instanceId="dialog" isOpen title="Title" onResolve={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with message and custom confirm button text', () => {
    const { asFragment } = render(
      <Dialog
        instanceId="dialog"
        isOpen
        title="Delete file"
        message="Are you sure?"
        confirmButtonText="Delete"
        onResolve={() => {}}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders without cancel button', () => {
    const { asFragment } = render(
      <Dialog instanceId="dialog" isOpen title="Only confirm" hideCancelButton onResolve={() => {}} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders in modal mode', () => {
    const { asFragment } = render(<Dialog instanceId="dialog" isOpen title="Modal" modal onResolve={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with custom header, content and footer', () => {
    const { asFragment } = render(
      <Dialog
        instanceId="dialog"
        isOpen
        title="Ignored title"
        header={<div data-testid="custom-header">Header</div>}
        content={<div data-testid="custom-content">Content</div>}
        footer={<div data-testid="custom-footer">Footer</div>}
        onResolve={() => {}}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
