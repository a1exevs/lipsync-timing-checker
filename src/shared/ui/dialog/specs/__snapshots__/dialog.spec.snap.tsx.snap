// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Dialog renders default with title only 1`] = `<DocumentFragment />`;

exports[`Dialog renders in modal mode 1`] = `<DocumentFragment />`;

exports[`Dialog renders with custom header, content and footer 1`] = `<DocumentFragment />`;

exports[`Dialog renders with message and custom confirm button text 1`] = `<DocumentFragment />`;

exports[`Dialog renders without cancel button 1`] = `<DocumentFragment />`;
